src/dispatch/incident/flows.py

```diff
@@ -914,6 +966,20 @@ def incident_assign_role_flow(assigner_email: str, incident_id: int, action: dic
         )
 
     if assignee_role == ParticipantRoleType.incident_commander:
+        # we get the incident document
+        incident_document = get_document(
+            db_session=db_session,
+            incident_id=incident_id,
+            resource_type=INCIDENT_RESOURCE_INVESTIGATION_DOCUMENT,
+        )
+
+        # update the incident document
+        document_updates = {
+            f"Incident Commander: {old_commander_name}": f"Incident Commander: {incident.commander.name}"
+        }
+        document_plugin = plugins.get(INCIDENT_PLUGIN_DOCUMENT_SLUG)
+        document_plugin.update(incident_document.id, document_updates, raw=True)
+
         # we update the conversation topic
         set_conversation_topic(incident)
```

Comment:
@kevgliss I'd suggest we do this before updating the doc and ticket to provide feedback to the user right away.